docs/concepts/concepts.md

```diff
@@ -0,0 +1,88 @@
+
+### Datasets
+
+Datasets (not to be confused with a [Spark Dataset](https://spark.apache.org/docs/3.1.3/api/scala/org/apache/spark/sql/Dataset.html)) in Prophecy are the union of two concepts:
+
+- **_WHAT_** your data looks like
+- **_HOW_** to read it or write it.
+
+To define a dataset, all you need is
+
+- **Type**: The type of data you are reading/writing like CSV, Parquet files or catalog tables.
+- **Location**: The location of your data. It could be a file path for CSV or a table name.
+- **Properties**: Properties consists of Schema which is the _shape_ of the data and some other attributes specific to the file format. For example, in case of CSV, you can give Column delimiter in additional attributes.
+
+![Dataset](img/dataset.png)
+
+## Gems
+
+If a Pipeline is a roadmap of the journey that your data will take from Start to Finish, the Gems are the stops to make along the way. Each Gem instance can be seen as just another Spark Transformation you perform on your data like Reformat, Join, Limit or Filter.
+Gems are categorised as below:
+
+| <div style={{'width':'100px'}}>Gem</div>   | Category                                                     | Description                                                                                                |
+| ------------------------------------------ | ------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------- |
+| ![Source and Target](img/SourceTarget.png) | [**Source and Target**](/low-code-spark/gems/source-target/) | The set of Gems that help with loading and saving data.                                                    |
+| ![Transform](img/Transform.png)            | [**Transform**](/low-code-spark/gems/transform/)             | The set of Gems that help with transforming data.                                                          |
+| ![Join and split](img/JoinSplit.png)       | [**Join and Split**](/low-code-spark/gems/join-split/)       | The set of Gems that help with the process of merging or splitting DataFrame(s) to create new DataFrame(s) |
+| ![Custom](img/Custom.png)                  | [**Custom**](/low-code-spark/gems/custom/)                   | Set of Gems that do not fall into the other categories.                                                    |
+| ![Subgraph](img/Subgraph.png)              | [**Subgraph**](/low-code-spark/gems/subgraph/)               | A Gem that can contain many other Gems within it.                                                          |
+
+Each Gem _instance_ comes with their own configuration and each produces its own block of output code.
+
+## Jobs
+
+Once you have developed a Spark data Pipeline using Prophecy, you will want to schedule it to run at some frequency. You can schedule just a single Pipeline or create a DAG of Pipelines. You can also add scripts to run in between these Pipelines.
```

Comment:
Users might not understand DAG directly, let's add an example image below it

Reply:
Added the image